packages/@aws-cdk/aws-dynamodb/lib/replica-provider.ts

```diff
@@ -27,10 +27,36 @@ export class ReplicaProvider extends NestedStack {
    */
   public static getOrCreate(scope: Construct, props: ReplicaProviderProps = {}) {
     const stack = Stack.of(scope);
-    const uid = '@aws-cdk/aws-dynamodb.ReplicaProvider';
+    const uid = this.getUid(stack);
     return stack.node.tryFindChild(uid) as ReplicaProvider ?? new ReplicaProvider(stack, uid, props);
   }
 
+  public static clearUids() {
+    this.getOrCreateCalls.clear();
+  }
+
+  // Map of getOrCreate() calls per stack
+  private static getOrCreateCalls = new Map<string, number>();
```

Comment:
Minor: Not sure if this is the best name for this. At first, I thought it is a method.

How about `replicasPerStackMap`?

Reply:
OK but then the right name would be `providersPerStack` (you can have multiple replicas with a single provider).

Reply:
I think it's the right name... you can have 1 table with 5 replicas in a stack and the counter would still be 1.